### ISSUE TITLE
ERROR: Service 'frontend' failed to build: The command '/bin/sh -c docker-php-source extract   && docker-php-source delete   && docker-php-ext-install json mysqli' returned a non-zero code: 2

### ISSUE THREAD

**Issue by jonahyongNTU** [dnorgaard-usgs/ffrsam#1]
Hi FFRSAM,

I am trying to install and setup FFRSAM on CentOS7. Is this possible?

I have docker running well and docker responds well to "docker info" and "docker run hello-world".

I followed your installation instructions and got to the part where I am supposed to do:
docker-compose build

It starts with building the backend and successfully completes it:
Successfully built d9d4a96da6a4

Then it tries to build the frontend and fails at Step 5/9:
Step 5/9 : RUN docker-php-source extract   && docker-php-source delete   && docker-php-ext-install json mysqli
 ---> Running in 4fc5de513a0c

Configuring for:
PHP Api Version:         20190902
Zend Module Api No:      20190902
Zend Extension Api No:   320190902

The error in red text says:
ls: cannot access '.': Operation not permitted
configure: error: working directory cannot be determined
ERROR: Service 'frontend' failed to build: The command '/bin/sh -c docker-php-source extract   && docker-php-source delete   && docker-php-ext-install json mysqli' returned a non-zero code: 2

I do not know where to begin looking to fix this problem.

Are you able to help me, please?

Thanks,
Jonah.
![ffrsam-build-error](https://user-images.githubusercontent.com/120165368/206648287-23d0c83e-1d71-4b65-b3a3-ecb970fd3f87.png)

**Comment by dlnorgaard** [comment]
What version of Docker are you running?  Can you make sure it is running the latest?  

Others are running into similar issue and one post reports that updating Docker fixed the problem:
https://stackoverflow.com/questions/68848154/failed-job-to-create-docker-image

Other options seem to be to try different images in frontend.Dockerfile, e.g.:

- php:7.3.28-apache
- php:7.4-apache-buster
- etc

**Comment by jonahyongNTU** [comment]
Oh! haha! I think I was running docker version 2 from the base CentOS repo.
Thank you very much, dlnorgaard, for pointing me towards docker version 20. I have upgraded mine to version 20 and it successfully solves the problem!

Just some extra info, for anyone who may encounter some obstacles:
The default package in the base CentOS repo is named "docker", which is different from the current version named "docker-ce".
I found these step-by-step instructions useful for upgrading my docker to the latest version 20.
https://docs.docker.com/engine/install/centos/
